### DISCUSSION TITLE
MLX 90640 Thermal camera usage

### DISCUSSION THREAD

**Discussion by erewhon** [AllskyTeam/allsky-modules#134] [General]
Hi there!   I recently installed AllSky on a Raspberry Pi, and decided to try out the MLX 90640 module.  It works great, and I'd like to use it in my allsky camera.   The only problem: how can I mount the sensor outside?   I tested it behind an acrylic dome, and it wasn't able to read through it.  (I didn't think it would work, which is why I tested it first.)

Is there any way to mount it where it can see the sky yet not get fried by rain?  I thought about drilling a hole in the enclosure and then caulking around the imaging part of the sensor keeping the rest inside.  But that still doesn't seem like it would be safe to expose it to the elements.

Thanks!

**Comment by Alex-developer** [comment]
I have mounted mine behind 'cling film' which seems to work ok

**Reply by erewhon** [reply]
Oh... that sounds like it would be good.  Do you have a hole drilled in a case so it pokes through?  I'm at the case modification stage, so just trying to work it out.   Thanks!

**Reply by Alex-developer** [reply]
> Oh... that sounds like it would be good.  Do you have a hole drilled in a case so it pokes through?  I'm at the case modification stage, so just trying to work it out.   Thanks!

All of my sensors are in separate cases and I use waterproof connectors to connect them to the main camera housing. This lets me position the sensors wherever I need them.

So for the mlx I drilled a hole in a small project box and mounted the sensor flush with the hole. I then covered the top of the project box in cling film. Every few weeks I check the cling film to make sure it's still intact

**Reply by erewhon** [reply]
Ok, cool.  That's actually very similar to the approach I was thinking of last night: putting the sensors in a separate box.  Thanks for the details!

**Comment by Alex-developer** [comment]
Did you manage to get this to work?
